[Jovita Anderson] iP

jovitaanderson opened the pull request:
## Ado
>“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes)) 

Ado frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER* FAST to use

All you need to do is,
1. download it from [here](https://github.com/jovitaanderson/ip/releases/tag/A-Jar).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

jovitaanderson committed:
Implemented basic skeletal version of Duke

jovitaanderson committed:
Add the ability to store text to list

jovitaanderson committed:
Add ability to mark tasks

jovitaanderson committed:
Add different Task types

jovitaanderson committed:
Updated EXPECTED.TXT, input.txt

jovitaanderson committed:
Add DukeException

jovitaanderson committed:
Add delete task

jovitaanderson committed:
Handle more errors

jovitaanderson committed:
Add save feature

jovitaanderson committed:
Add date feature to deadline

jovitaanderson committed:
Merge branch 'branch-Level-7'

jovitaanderson committed:
Merge branch 'branch-Level-8'

jovitaanderson committed:
Modify code structure

jovitaanderson committed:
Organize classes into packages

jovitaanderson committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

jovitaanderson committed:
Add gradle, JUnit tests

jovitaanderson committed:
Package app as a JAR file

jovitaanderson committed:
Add JavaDoc

jovitaanderson committed:
Edit to coding standard

jovitaanderson committed:
Add Find feature

jovitaanderson committed:
Merge branch 'A-CodingStandard' into Level-9

jovitaanderson committed:
Merge branch 'branch-A-JavaDoc' into Level-9

# Conflicts:
#	src/main/java/duke/Ui.java

jovitaanderson committed:
Edit to coding standard

JordanChua commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,55 @@
+            return new ListCommand();
+        case "bye":
+            return new ByeCommand();
+        case "todo":
```
Perhaps it would be good to separate the statements into various sections, for example organising the tasks into a separate paragraph, then leaving a line and grouping the mark/unmark operations into another section. 

Adding a few comment headers would also be useful, such as how the input format of the date and time for a deadline is input since it may vary from user to user

jovitaanderson replied:
Thanks for the tip!

JordanChua commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+package duke;
+
+import java.time.LocalDate;
```
Here the import statements can be separated to determine the dependencies more easily

jovitaanderson replied:
Hmm, separating the import statments seems to violate the checkstyle 🤔

JordanChua commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+     * @param text keyword to find in task description.
+     * @return matching tasks in string form.
+     */
+    public String listToStringWithText(List<Task> list, String text) {
```
perhaps a more intuitive method name would be good, for example "matchingTasklistToString"

jovitaanderson replied:
Thanks!

JordanChua commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+        System.out.println(message);
+    }
+
+    void validate(String input) throws DukeException {
```
Adding a javadoc comment would be useful here since the method body is relatively long

JordanChua commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+            //handles invalid commands
+            throw new DukeException(mainCommand + "? (´･_･`) I don't know what that means\n");
+        }
+        String[] commandsWithDescription = {"todo", "deadline", "event", "find"};
```
adding a comment header and leaving a blank line would help as well, so that the reader understands that this handles the scenario where certain commands without a body are meaningless and hence should throw an error. 

Perhaps other commands such as mark/unmark without a integer attached to it could be meaningful as well

JordanChua commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,136 @@
+                break;
+            case "deadline":
+                String[] deadlineSegments = commandSegments[1].split("/by", 2);
+                if (deadlineSegments.length < 2) {
```
leaving a blank line between the sections would be useful since we are handling various input mismatch here, and adding a comment for each specific one would be good as well:
1. if no date is input
2. a date is input but the date format is incorrect

JordanChua commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,123 @@
+                String taskType = taskSegment[0];
+                String taskStatus = taskSegment[1];
+                String taskDescription = taskSegment[2];
+                String taskDate = null;
```
perhaps leaving a blank line here and adding a comment header would be good here to indicate that the taskdate is meant for events, and taskLocalDate is meant for deadlines.

JordanChua reviewed (commented):
Overall looks good :)
Just a few minor sections where comment headers and spacing could be useful :)

jovitaanderson committed:
Change text UI to GUI

jovitaanderson committed:
Update gradle

alexandermula commented on data/tasks.txt:
```diff
@@ -0,0 +1,4 @@
+T|X|readbook
+E| |meeting | Mon 10pm
+D| |do homework |2022-08-25
+E|X|feed cat | sun 3pm
```
Could consider including data/tasks.txt in gitignore.

alexandermula commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+            System.out.println("list size-" + tasks.listSize());
```
Maybe can remove this console log if it's for debugging purposes.

jovitaanderson replied:
Oh i forgot about that, thanks!

alexandermula commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,55 @@
+     * @throws DukeException If command does not exist.
+     */
+    public static Command parse(String fullCommand) throws DukeException {
+        String[] commandSegments = fullCommand.split(" ", 2);
+        String mainCommand = commandSegments[0].toLowerCase().trim();
```
Clean way of splitting fullCommand. Helps with simpler code in switch statements:)

alexandermula commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+        return "[D]" + super.toString() + " (by:" + formattedDate + ")";
+    }
+
+    @Override
+    public String toStringForStorage() {
+        return "D|" + super.toStringForStorage() + "|" + date.toString();
+    }
```
Really useful toString for saving to storage easily.

alexandermula reviewed (commented):
LGTM! Overall seems to be following OOP concepts and is user-friendly. Good Job:)

NicholasPatrick commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,168 @@
+        //printMessage(startMessage);
+    }
+
+    String readCommand() throws DukeException {
+        Scanner sc = new Scanner(System.in);
+        String input = sc.nextLine();
+        validate(input.toLowerCase());
+        return input;
+    }
+
+    /**
+     * Prints out partition line to user interface.
+     */
+    public void showLine() {
+        System.out.println(PARTITION);
+    }
```
Since all `Ui` objects read from `System.in` and writes to `System.out`, have you considered making these methods (and perhaps the other methods in this class) `static`? This way, you don't need to create a `Ui` object.

NicholasPatrick commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,55 @@
+        case "todo":
+            return new TodoCommand(commandSegments[1].trim());
+        case "deadline":
+            String[] deadlineSegments = commandSegments[1].split("/by", 2);
+            return new DeadlineCommand(deadlineSegments[0], LocalDate.parse(deadlineSegments[1].trim()));
+        case "event":
+            String[] eventSegments = commandSegments[1].split("/at", 2);
+            return new EventCommand(eventSegments[0], eventSegments[1]);
```
The deadline with one date specified and the event with two dates specified are treated differently here. Have you considered splitting the two dates for Event or not parsing the date before instantiating a `DeadlineCommand`?

NicholasPatrick commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,168 @@
+ * Handles the interactions with the user.
+ */
+public class Ui {
+    static final String CHATBOX_NAME = "Ado";
```
Looks like you forgot to use or delete this constant.

jovitaanderson replied:
Oh, thanks for highlighting! :)

NicholasPatrick commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,63 @@
+/**
+ * Represents the main functions of Duke chatbot and initializes require variables.
+ */
+public class Duke {
```
Have you considered renaming this class to `Ado`?

NicholasPatrick reviewed (commented):
Looks like you've done great OOP. Just needs a few minor tweaks here and there.

jovitaanderson committed:
Add assertion to Parser, Storage and Ui

There are no tests implemented to ensure that assumptions made
about the program are correct.

Adding assertions allows testing for the correctness of any
assumptions that have been made in the program

Let's update the Parser, Ui, and Storage script to include
assertions

Assertions are added to test the preconditions and postconditions
of the method to ensure it is working as expected

jovitaanderson committed:
Improve Code Quality

Code is not easily readable, and has some use of deep nesting as
well as magic strings.

Code needs to be concise and readable so that other developers
can understand and modify the code easily

Let's
* improve readability by using boolean to represent the required
state of if/else loops
* avoid deep nesting by using  if/else if  instead of nested if's
statements
* avoid magic strings by implementing named constants

jovitaanderson committed:
Use streams in Ui script

jovitaanderson committed:
Merge pull request #1 from jovitaanderson/A-CodeQuality

jovitaanderson committed:
Merge pull request #2 from jovitaanderson/A-Assertions

Add assertion to Parser, Storage and Ui

jovitaanderson committed:
Merge branch 'master' of https://github.com/jovitaanderson/ip into A-Streams

# Conflicts:
#	src/main/java/duke/Ui.java

jovitaanderson committed:
Merge pull request #3 from jovitaanderson/A-Streams

jovitaanderson committed:
Update stream in Ui script

jovitaanderson committed:
Add help command and sample data if user is new

jovitaanderson committed:
Merge pull request #4 from jovitaanderson/C-Help

C help

jovitaanderson committed:
Modify code based on feedback given

Some feedback where given by peers to improve readability
 and remove unused code

Let's
* add comments to understand usage for variables
* add line space to improve readability
* remove unused code
* add data.txt to .gitignore

jovitaanderson committed:
Update GUI design

jovitaanderson committed:
Update code to follow coding standard, edit chatbot output messages

jovitaanderson committed:
Improve code to handle more errors and  add some Junit tests

jovitaanderson committed:
Update sample data

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Create _config.yml

jovitaanderson committed:
Delete _config.yml

jovitaanderson committed:
Create _config.yml

jovitaanderson committed:
Rename all Duke class,variables,packages to Ado

jovitaanderson committed:
Merge branch 'master' of https://github.com/jovitaanderson/ip

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update tasklist to save after every command, program exits on 'bye'

jovitaanderson committed:
Merge branch 'master' of https://github.com/jovitaanderson/ip

jovitaanderson committed:
Add images for README

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md JAR link

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Fix README.md emoji

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Fix Table

jovitaanderson committed:
Fix typo

jovitaanderson committed:
Fix typo

jovitaanderson committed:
Update help message

jovitaanderson committed:
Merge branch 'master' of https://github.com/jovitaanderson/ip

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Improve code quailty

jovitaanderson committed:
Update images

jovitaanderson committed:
Update images

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md

jovitaanderson committed:
Update README.md
